script/windows/14.0/update/WindowsUpdateCheck.php

```diff
@@ -51,22 +54,50 @@ private function BuildKeymanDesktopVersionResponse($tier, $InstalledVersion, $re
       // This gets us upgrades on alpha, beta and stable tiers.
       $tiers = get_object_vars($this->DownloadVersions->windows);
 
-      return $this->CheckVersionResponse($tier, $tiers, $InstalledVersion, $regex);
+      // We will support staying on alpha tier even once a version
+      // hits stable. This is correct, with major upgrade support, as
+      // that'll mean the user will transition to a new version on the
+      // same tier (i.e. always staying bleeding edge). However, for beta
+      // users, the assumption will be that they should upgrade to the stable
+      // release once it is available, as the beta branch will go stale and
+      // they will receive no further updates on it otherwise (until next beta
+      // period, anyway).
+      switch($tier) {
+        case 'alpha':
+          return $this->CheckVersionResponse($tier, $tiers, $InstalledVersion, $regex);
+        case 'beta':
+          $response = $this->CheckVersionResponse($tier, $tiers, $InstalledVersion, $regex);
+          if($response === FALSE)
+            $response = $this->CheckVersionResponse('stable', $tiers, $InstalledVersion, $regex);
+          return $response;
+        case 'stable':
+          return $this->CheckVersionResponse($tier, $tiers, $InstalledVersion, $regex);
+        default:
+          fail('Unexpected tier '.$tier, 500);
+      }
     }
 
     private function CheckVersionResponse($tier, $tiers, $InstalledVersion, $regex) {
       if(!isset($tiers[$tier])) return FALSE;
       $tierdata = $tiers[$tier];
 
-      // We will support staying on alpha or beta tier once a version
-      // hits stable. This is correct, with major upgrade support, as
-      // that'll mean the user will transition to a new version on the
-      // same tier (i.e. always staying bleeding edge)
-
       $files = get_object_vars($tierdata->files);
       foreach($files as $file => $filedata) {
         // This is currently tied to Windows -- for other platforms we need to change this
         if(preg_match($regex, $file)) {
+
+          if(!$this->isManual &&
```

Comment:
If my read of this is correct... any user that _manually_ triggered the update check will bypass this 'staggered update' filter?

Just checking; I was actually about to suggest it before I realized you'd already put this in place.

Reply:
Correct